backend/src/main/java/com/tourneynizer/tourneynizer/controller/MatchController.java

```diff
@@ -14,16 +16,22 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 @Controller("MatchController")
 public class MatchController {
 
     private final MatchService matchService;
+    private final SessionService sessionService;
 
     @Autowired
-    public MatchController(MatchService matchService) {
+    public MatchController(MatchService matchService, SessionService sessionService) {
         this.matchService = matchService;
+        this.sessionService = sessionService;
     }
 
     @GetMapping("/api/tournament/{id}/match/getAll")
```

Comment:
Can you get rid of the cookie requirement here while you're at it please

Reply:
No, that's another PR

Reply:
fine